pyfarm/models/job.py

```diff
@@ -547,6 +547,36 @@ def alter_frame_range(self, start, end, by):
             if self.state != WorkState.RUNNING:
                 self.state = None
 
+    def rerun(self):
+        running_tasks = False
+        for task in self.tasks:
+            if task.state != _WorkState.RUNNING and task.state is not None:
+                task.state = None
+                task.agent = None
+                task.failures = 0
+                db.session.add(task)
+            elif task.state == _WorkState.RUNNING or task.agent is not None:
+                running_tasks = True
+
+        if not running_tasks:
+            self.state = None
+        else:
+            self.state = WorkState.RUNNING
+        self.completion_notify_sent = False
+        db.session.add(self)
+
+    def rerun_failed(self):
```

Comment:
Small doc string would be nice so you can tell what the method does on readthedocs.